torchvision/prototype/models/vision_transformer.py

```diff
@@ -230,6 +231,69 @@ def forward(self, x: torch.Tensor):
+        n, seq_length, hidden_dim = pos_embedding.shape
+        if n != 1:
+            raise ValueError(f"Unexpected position embedding shape: {pos_embedding.shape}")
+        if hidden_dim != self.hidden_dim:
+            raise ValueError(f"Position embedding hidden_dim incorrect: {hidden_dim}" f", expected: {self.hidden_dim}")
+        new_seq_length = self.seq_length
+
+        # Need to interpolate the weights for the position embedding.
+        # We do this by reshaping the positions embeddings to a 2d grid, performing
+        # an interpolation in the (h, w) space and then reshaping back to a 1d grid.
+        if new_seq_length != seq_length:
+            # The class token embedding shouldn't be interpolated so we split it up.
+            seq_length -= 1
+            new_seq_length -= 1
+            pos_embedding_token = pos_embedding[:, :1, :]
+            pos_embedding_img = pos_embedding[:, 1:, :]
+
+            # (1, seq_length, hidden_dim) -> (1, hidden_dim, seq_length)
+            pos_embedding_img = pos_embedding_img.permute(0, 2, 1)
+            seq_length_1d = int(math.sqrt(seq_length))
+            torch._assert(seq_length_1d * seq_length_1d == seq_length, "seq_length is not a perfect square!")
+
+            logging.info(
+                "Interpolating the position embeddings from image "
+                f"{seq_length_1d * self.patch_size} to size {self.image_size}"
+            )
+
+            # (1, hidden_dim, seq_length) -> (1, hidden_dim, seq_l_1d, seq_l_1d)
+            pos_embedding_img = pos_embedding_img.reshape(1, hidden_dim, seq_length_1d, seq_length_1d)
+            new_seq_length_1d = self.image_size // self.patch_size
+
+            # Use bicubic interpolation.
+            # (1, hidden_dim, seq_l_1d, seq_l_1d) -> (1, hidden_dim, new_seq_l_1d, new_seq_l_1d)
+            new_pos_embedding_img = nn.functional.interpolate(
+                pos_embedding_img,
+                size=new_seq_length_1d,
+                mode="bicubic",
+                align_corners=True,
+            )
+
+            # (1, hidden_dim, new_seq_l_1d, new_seq_l_1d) -> (1, hidden_dim, new_seq_length)
+            new_pos_embedding_img = new_pos_embedding_img.reshape(1, hidden_dim, new_seq_length)
+
+            # (1, hidden_dim, new_seq_length) -> (1, new_seq_length, hidden_dim)
+            new_pos_embedding_img = new_pos_embedding_img.permute(0, 2, 1)
+            new_pos_embedding = torch.cat([pos_embedding_token, new_pos_embedding_img], dim=1)
+
+            model_state["encoder.pos_embedding"] = new_pos_embedding
+
+            if reset_heads:
+                model_state_copy = {}
+                for k, v in model_state.items():
+                    if not k.startswith("heads"):
+                        model_state_copy[k] = v
+                model_state = model_state_copy
+
+            self.load_state_dict(model_state, strict=strict)
+
+            logging.info("Chekpoint loaded.")
```

Comment:
We are not currently using `logging` anywhere else. I would suggest removing the debug messages.